projects/igniteui-angular/src/lib/expansion-panel/expansion-panel.directives.ts

```diff
@@ -1,4 +1,36 @@
-import { Directive, HostBinding } from '@angular/core';
+import { Directive, ElementRef, HostBinding } from '@angular/core';
+
+/**
+ * Returns the `textContent` of an element
+ *
+ * ```html
+ * <igx-expansion-panel-title>
+ *  Tooltip content
+ * </igx-expansion-panel-title>
+ * ```
+ *
+ *  or the `title` content
+ *
+ * ```html
+ * <igx-expansion-panel-title [title]="'Tooltip content'">
+ * </igx-expansion-panel-title>
+ * ```
+ *
+ * If both are provided, returns the `title` content.
+ *
+ * @param element
+ * @returns tooltip content for an element
+ */
+const getTooltipContent = (element: ElementRef): string => {
```

Comment:
Create a base class with this method and extend both directives from this new class.